packages/core/src/utils/path.ts

```diff
@@ -115,7 +115,14 @@ export function setInPath(object: NestedRecord, path: string, value: unknown): v
   for (let i = 0; i < keys.length; i++) {
     // Last key, set it
     if (i === keys.length - 1) {
-      acc[keys[i]] = value;
+      const targetKey = keys[i];
+
+      if (typeof value === 'boolean' && typeof acc[targetKey] === 'object' && acc[targetKey] !== null) {
+        setAllChildrenToValue(acc[targetKey] as Record<string, unknown>, value);
+      } else {
+        acc[targetKey] = value;
+      }
```

Comment:
This might cause a bug since we use this exact function for form values setting, and let's say a path might be an object or a boolean, this will break it as it will set all the children to boolean rather than set the exact path to a boolean.

We probably should split it out in a separate function that is only used by `touched` mutators and not the value mutators.

Reply:
do you suggest have a specialized method in `path.ts` called something like `setTouchedInPath` and update the `setFieldTouched` in the `formContetx.ts`?

```ts
 function setFieldTouched<TPath extends Path<TForm>>(path: TPath, value: boolean) {
    // setInPath(touched, path, value);
    setTouchedInPath(touched, path, value);
  }
  ```
  
  Although, in my testing, I didn't see any bugs come up in my original implementation... 🤔  Do you know some edge cases that I can try that would cause a bug?

Reply:
It would pop up in discriminated union objects where you want to se the exact path to the value and not its nested children.

For example a Form with:

```ts
z.object({
  someConfig: z.object({
    // Some deep configs
  }).or(z.boolean())
})
```

So, yes a specialized version of this could be useful. You can probably do something similar like we did with unsetPath, by passing a setter, so the touched setter would have that nested portion while the default is just the plain set.